qrexec/tests/socket/qrexec.py

```diff
@@ -93,12 +94,14 @@ def recv_all_messages(self):
             messages.append((message_type, data))
         return messages
 
-    def handshake(self):
-        self.send_message(MSG_HELLO, struct.pack("<L", QREXEC_PROTOCOL_VERSION))
+    def handshake(self, protocol_version=None):
+        if not protocol_version:
+            protocol_version = QREXEC_CLIENT_PROTOCOL_VERSION
```

Comment:
Wouldn't it make more sense to do the other way around? Default to version QREXEC_PROTOCOL_VERSION (set as 4?), only in cases where older version is required provide the parameter. I think you'll need to use the parameter in less places then.

And also, there is no need for `=None` and then the if - simply put default value in the function header.

Reply:
Yes multiple iterations and removing of code. I did it the way around.